plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py

```diff
@@ -1587,40 +1587,43 @@ def _try_get_wealth(address: Address, url: Optional[str] = None) -> None:
             )
 
 
-class LruLockWrapper:
-    """Wrapper for LRU with threading.Lock."""
+class SessionCacheLockWrapper:
```

Comment:
The version we're using does not use [`lru.LRU`](https://github.com/ethereum/web3.py/blob/v5.25.0/web3/_utils/request.py#L34) for session cache anymore instead it uses a [custom cache object](https://github.com/ethereum/web3.py/blob/v5.31.4/web3/_utils/request.py#L68)